### PR TITLE
Add script to check if drug alerts have been emailed

### PR DESCRIPTION
**Pull request by tijmenb**
GOV.UK provides email alerts. This repo provides a script that reports if emails have been sent for all [drug and medical advice alerts](https://www.gov.uk/drug-device-alerts). 

See the readme for more information and usage.

Trello: https://trello.com/c/GGGYpRXd

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+task :run do
+  require_relative './lib/drug_alert_email_verifier'
+
+  verifier = DrugAlertEmailVerifier.new
+
+  unless verifier.have_all_alerts_been_emailed?
```
Comment (Davidslv) [inline Rakefile]: maybe you could add a message saying that "all alerts have been emailed" ?

File: lib/drug_alert_email_verifier.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+require_relative './drug_alerts'
+
+class DrugAlertEmailVerifier
+  attr_reader :missing_alerts
+
+  EMAILS_THAT_SHOULD_HAVE_RECEIVED_ALERT = %w[
+    googleapi+immediately@digital.cabinet-office.gov.uk
+    googleapi+daily@digital.cabinet-office.gov.uk
+    googleapi+weekly@digital.cabinet-office.gov.uk
+  ]
```
Comment (Davidslv) [inline lib/drug_alert_email_verifier.rb]: is it ok to have emails exposed in a public repo?

**Comment by MatMoore** [comment]
I might be misunderstanding something, but will this create false positives for the weekly and daily emails? It looks like the alert just has to be an hour old for it to be expected in all inboxes.

**Comment by tijmenb** [comment]
@Davidslv good point about the emails (it's not a risk but feels better to get them in a env var). I've removed your comment to let GitHub forget about the old commit.

**Comment by Davidslv** [comment]
MatMoore, it does the opposite, it doesn't care about the last hour to prevent false positives, but it does check before the last hour. https://github.com/alphagov/email-alert-monitoring/pull/1/files#diff-607f6635994f6884b21f7711264f54d9R16

**Comment by tijmenb** [comment]
@MatMoore that's a good one to clarify for me. The drug alerts are so important that they have a special "urgent" flag in govdelivery (I don't know where this is documented or set). This means that even if somebody has requested a weekly digested email for these alerts, they should be sent immediately. That's the reason this check tests with a weekly digest, daily digest and immediately subscribed users.

I'll add some docs to clarify this all.

File: Rakefile (hunk 1 above)
Comment (tijmenb) [inline Rakefile]: Yes, good idea.

**Comment by tijmenb** [comment]
@MatMoore I've attempted to [explain it here](https://github.com/alphagov/email-alert-monitoring/pull/1/files#diff-04c6e90faac2675aa89e2176d2eec7d8R13). Can you see if it makes sense?

**Comment by MatMoore** [comment]
@tijmenb ah, of course - yes that makes sense :+1:

File: lib/google_auth.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    authorizer = Google::Auth::UserAuthorizer.new(
+      client_id,
+      Google::Apis::GmailV1::AUTH_GMAIL_READONLY,
+      EnvironmentTokenStore.new
+    )
+
+    authorizer.get_credentials('default')
+  end
+
+  class EnvironmentTokenStore
```
Comment (mobaig) [inline lib/google_auth.rb]: A little nitpicky, but perhaps `DummyEnvironmentTokenStore` might be better given how little this object does.

File: spec/feautures/drug_alert_email_verification_spec.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
```
Comment (mobaig) [inline spec/feautures/drug_alert_email_verification_spec.rb]: typo in filename

**Comment by mobaig** [comment]
some minor stuff but otherwise LG

**Comment by tijmenb** [comment]
@mobaig updated

**Pull request merged**